typed-racket-test/succeed/define-new-subtype-poly.rkt

```diff
@@ -0,0 +1,13 @@
+#lang typed/racket/base
+
+(define-new-subtype (Meters a) (meters a))
+
+(: m+ : (case-> [(Meters Real) (Meters Real) -> (Meters Real)]
+                [(Meters Number) (Meters Number) -> (Meters Number)]))
+(define (m+ a b)
+  (meters (+ a b)))
+
+(define x1 (m+ (meters 1) (meters 2)))
+(ann x1 (Meters Real))
+(define x2 (m+ (meters (ann 1 Number)) (meters 2)))
+(ann x2 (Meters Number))
```

Comment:
1. Can any of these tests be unit tests instead of integration tests? (i.e. in `typecheck-tests.rkt` perhaps)
2. There are no tests for this new feature failing when misused in any way -- I worry we can't be confident if this implementation is right without positive and negative tests. Are there any sensible tests of that variety we can add to this PR?

Reply:
You're right, I should add negative tests. However, I'm a bit busy now, and this pull request can wait. Especially if we plan on replacing Distinction with intersection-based types like you've talked about before.